src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1786,6 +1786,36 @@ internal PointerTypeSymbol CreatePointerTypeSymbol(TypeSymbol elementType)
             return new PointerTypeSymbol(elementType);
         }
 
+        private protected override bool IsSymbolAccessibleWithinCore(
+            ISymbol symbol,
+            ISymbol within,
+            ITypeSymbol throughType)
+        {
+            var symbol0 = symbol as Symbol;
+            if (symbol0 is null)
+            {
+                throw new ArgumentException(string.Format(CodeAnalysisResources.IsSymbolAccessibleWrongAssembly, nameof(symbol)), nameof(symbol));
```

Comment:
> throw new ArgumentException(string.Format(CodeAnalysisResources.IsSymbolAccessibleWrongAssembly, nameof(symbol)), nameof(symbol)); [](start = 15, length = 131)

I think we should use the same helper that we use in other similar APIs that take interfaces for symbols - EnsureCSharpSymbolOrNull. It will also throw an exception with more precise message.